### PR TITLE
feat(dns): support get list of the all name servers

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

Add a new datasource supports getting list of the all name servers.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
 make testacc TEST=./huaweicloud/services/acceptance/dns TESTARGS='-run TestAccDataSourceNameservers_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dns -v -run TestAccDataSourceNameservers_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceNameservers_basic
=== PAUSE TestAccDataSourceNameservers_basic
=== CONT  TestAccDataSourceNameservers_basic
--- PASS: TestAccDataSourceNameservers_basic (23.17s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dns       23.230s
```

### CODE REVIEW

File: docs/data-sources/dns_nameservers.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+The `ns_records` block supports:
+
+* `hostname` - The host name of the public name server.
+  If `type` is set to `private`, the value is an empty string.
+
+* `address` - The  address of the private name server.
+  If `type` is set to `public`, the value is an empty string.
+
+* `priority` - The priority of  the name server.
+  The smaller the value, the higher the priority.
```
Comment (Lance52259) [inline docs/data-sources/dns_nameservers.md]:
```suggestion
  The smaller value means the higher priority.
```

File: docs/data-sources/dns_nameservers.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `type` - (Optional, String) Specifies the type of the name server.
+  The valid values are as follows:
+  + **public**: indicates a public name server.
+  + **private**: indicates a private name server.
```
Comment (Lance52259) [inline docs/data-sources/dns_nameservers.md]:
```suggestion
* `type` - (Optional, String) Specifies the type of the name server.
  The valid values are as follows:
  + **public**
  + **private**
```
The description seems incorrect

File: docs/data-sources/dns_nameservers.md (hunk 1 above)
Comment (wuzhuanhong) [reply docs/data-sources/dns_nameservers.md]: fixed

File: docs/data-sources/dns_nameservers.md (hunk 2 above)
Comment (wuzhuanhong) [reply docs/data-sources/dns_nameservers.md]: fixed

**Pull request merged**